packages/core/content-releases/admin/src/components/AddToReleaseAction.tsx

```diff
@@ -0,0 +1,299 @@
+export const RELEASE_ACTION_FORM_SCHEMA = yup.object().shape({
+  type: yup.string().oneOf(['publish', 'unpublish']).required(),
+  releaseId: yup.string().required(),
+});
+
+export interface FormValues {
+  type: CreateReleaseAction.Request['body']['type'];
+  releaseId: CreateReleaseAction.Request['params']['releaseId'];
+}
+
+export const INITIAL_VALUES = {
+  type: 'publish',
+  releaseId: '',
+} satisfies FormValues;
+
+interface AddActionToReleaseModalProps {
+  contentTypeUid: string;
+  documentId?: string;
+  onInputChange: (field: keyof FormValues, value: string | number) => void;
+  values: FormValues;
+}
+
+export const NoReleases = () => {
+  const { formatMessage } = useIntl();
+  return (
+    <EmptyStateLayout
+      icon={<EmptyDocuments width="16rem" />}
+      content={formatMessage({
+        id: 'content-releases.content-manager-edit-view.add-to-release.no-releases-message',
+        defaultMessage:
+          'No available releases. Open the list of releases and create a new one from there.',
+      })}
+      action={
+        <LinkButton
+          to={{
+            pathname: '/plugins/content-releases',
+          }}
+          tag={ReactRouterLink}
+          variant="secondary"
+        >
+          {formatMessage({
+            id: 'content-releases.content-manager-edit-view.add-to-release.redirect-button',
+            defaultMessage: 'Open the list of releases',
+          })}
+        </LinkButton>
+      }
+      shadow="none"
+    />
+  );
+};
+
+const AddActionToReleaseModal = ({
+  contentTypeUid,
+  documentId,
+  onInputChange,
+  values,
+}: AddActionToReleaseModalProps) => {
+  const { formatMessage } = useIntl();
+  const [{ query }] = useQueryParams<{ plugins?: { i18n?: { locale?: string } } }>();
+  const locale = query.plugins?.i18n?.locale;
+
+  // Get all 'pending' releases that do not have the entry attached
+  const response = useGetReleasesForEntryQuery({
+    contentTypeUid,
+    documentId,
+    hasEntryAttached: false,
+    locale,
+  });
+
+  const releases = response.data?.data;
+
+  if (releases?.length === 0) {
+    <NoReleases />;
+  }
+
+  return (
+    <Flex direction="column" alignItems="stretch" gap={2}>
+      <Box paddingBottom={6}>
+        <Field.Root required>
+          <Field.Label>
+            {formatMessage({
+              id: 'content-releases.content-manager-edit-view.add-to-release.select-label',
+              defaultMessage: 'Select a release',
+            })}
+          </Field.Label>
+          <SingleSelect
+            required
+            placeholder={formatMessage({
+              id: 'content-releases.content-manager-edit-view.add-to-release.select-placeholder',
+              defaultMessage: 'Select',
+            })}
+            name="releaseId"
+            onChange={(value) => onInputChange('releaseId', value)}
+            value={values.releaseId}
+          >
+            {releases?.map((release: any) => (
```

Comment:
Doesn't the type come throught? I think you might be able to get rid of this `any`